quixstreams/models/serializers/avro.py

```diff
@@ -0,0 +1,58 @@
+from typing import Union, Mapping, Optional, Any, Iterable
+
+from io import BytesIO
+
+from fastavro import schemaless_reader, schemaless_writer, parse_schema
+from fastavro.types import Schema
+
+from .base import Serializer, Deserializer, SerializationContext
+from .exceptions import SerializationError
+
+__all__ = ("AvroSerializer", "AvroDeserializer")
+
+
+class AvroSerializer(Serializer):
+    def __init__(self, schema: Schema, **kwargs):
+        """
+        Serializer that returns data in Avro format.
+
+        Additional keyword arguments are passed directly to fastavro [schemaless_writer](https://fastavro.readthedocs.io/en/latest/writer.html#fastavro._write_py.schemaless_writer) method.
+
+        :param schema: The avro schema.
+        """
+        self._schema = parse_schema(schema)
+        self._kwargs = kwargs or {}
+
+    def __call__(self, value: Any, ctx: SerializationContext) -> bytes:
+        data = BytesIO()
+
+        try:
+            schemaless_writer(data, self._schema, value, self._kwargs)
+        except ValueError as exc:
+            raise SerializationError(str(exc)) from exc
+
+        result = data.getvalue()
+        data.close()
```

Comment:
Can the `BytesIO` be used as a context manager here? 
So we don't need to close() it manually.